pkg/block/fetcher.go

```diff
@@ -217,14 +217,45 @@ func (f *BaseFetcher) loadMeta(ctx context.Context, id ulid.ULID) (*metadata.Met
 		return nil, ErrorSyncMetaNotFound
 	}
 
+	writeMetaFunc := func(m *metadata.Meta) {
```

Comment:
I think the `meta.json` should never be rewritten (we should treat each object in the bucket as immutable). We should just read the LastModified and just save it in the in-memory Meta structure.

Reply:
I thought having lastModifiedTime in `meta.json` saves 1 obj store attributes call if we lose the in-memory Meta structure (restart, etc). Rewritten this also ensures all `meta.json` files have lastModifiedTime field including the old meta files.

I will update the pr as you suggested.